### DISCUSSION TITLE
How to store sessions in the database with Credentials Provider?

### DISCUSSION THREAD

**Discussion by aland-sleman** [nextauthjs/next-auth#2021] [Help]
## Description 🐜
i implemented user sign in via google and everything works fine when user signs user session gets stored in the database.
and if i want to i can remove the session to logout the user... now the user has to sign in again.

Now i want to implement the same thing but with the Credentials Provider...
and i got this error: [https://next-auth.js.org/errors#callback_credentials_jwt_error](url)

i fixed it by enabling JWT while on the page i noticed this: "If you are using a Credentials Provider, NextAuth.js will not persist users or sessions in a database - user accounts used with the Credentials Provider must be created and manged outside of NextAuth.js."
So what i understand is that  i can't store session in the database with the Credentials Provider...Why is that?

So now this is becoming a big problem for me and i really can't progress developing my app...

Is  there any way for me to store the session with Credentials Provider? 
google signin is not really a priority to me i just thought it would be nice if the user can sign in with google.

i understand that with the Credentials Provider i have to signup the users then when user signins in i have to authorize the request... i already done that... everything is fine expect for the sessions not getting stored.

Thanks for answering and sorry for my bad english

## How to reproduce ☕️

I'm using next-auth with Postgres without adapters
I've setup postgresql database with this Schema:
[https://next-auth.js.org/schemas/postgres](https://next-auth.js.org/schemas/postgres)
And i'm using this example:
[https://github.com/nextauthjs/next-auth-example](https://github.com/nextauthjs/next-auth-example)

**Comment by balazsorban44** [comment]
Please read the docs:

> It is intended to support use cases where you have an existing system you need to authenticate users against.

> The functionality provided for credentials based authentication is intentionally limited to discourage use of passwords due to the inherent security risks associated with them and the additional complexity associated with supporting usernames and passwords.

https://next-auth.js.org/providers/credentials
